### PR TITLE
mitmproxy: change source url

### PR DESCRIPTION
**Pull request by colemickens**
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by colemickens** [comment]
Addresses: #15041

**Comment by colemickens** [comment]
cc @FRidh

**Comment by FRidh** [comment]
In the future please use the attribute name (`pythonPackages.mitmproxy`)

**Comment by colemickens** [comment]
@FRidh Do you mean in the commit message? If so, I will keep that in mind for the future. Thanks.

**Comment by copumpkin** [comment]
Isn't `fetchurl` against a GitHub archive not very stable? Why not `fetchFromGitHub`, which handles the potentially nondeterministic archives for you?

**Comment by colemickens** [comment]
@copumpkin Can you elaborate? If I set the rev to the version tag, isn't it going to be just as unstable as the archive download? Or am I meant to use the current actual git revision for a given tag as `rev` which would ensure consistent source even if the tag is force updated later?

**Comment by copumpkin** [comment]
Sorry, I just mean that we have an existing abstraction that combines `fetchzip` with GitHub archive URLs called `fetchFromGitHub`. The reason we use `fetchzip` instead of `fetchurl` is that the former hashes the _contents_ of the tarball rather than the tarball itself, and GitHub (appears to) generate those `https://github.com/mitmproxy/mitmproxy/archive/blah` URLs periodically in a cache. Every time the cache gets flushed, GitHub will generate the tarball again, but it'll (sometimes?) have a different hash, even though the contents are identical. `fetchFromGitHub` deals with that nonsense for you.

Basically, I think `fetchurl` from `https://github.com/X/Y/archive/` is almost always going to fail after a while, unless GitHub changes their tarball generation code.

**Comment by colemickens** [comment]
Thanks for spelling it out; makes sense. Fixed with: #15331

**Comment by FRidh** [comment]
@copumpkin it is my understanding that that issue has been fixed. On GitHub you have releases and tags and the issue used to be with one of them.

**Comment by copumpkin** [comment]
Hmm, perhaps, but still seems using the abstraction where possible, so people don't need to really think about the implementation details.
